controller/src/test/java/org/jboss/as/controller/test/ReadFeatureDescriptionTestCase.java

```diff
@@ -0,0 +1,623 @@
+
+        // subsystem params
+        Map<String, ModelNode> params = extractParamsToMap(feature);
+        assertFeatureIdParam(SUBSYSTEM, TEST_SUBSYSTEM, params);
+        assertParamWithDefault(EXTENSION, TEST_EXTENSION, params);
+
+        // subsystem children
+        ModelNode resource1 = feature.require(CHILDREN).require("subsystem.testsubsystem.test.special-names-resource");
+        Assert.assertEquals("subsystem.testsubsystem.test.special-names-resource", resource1.get(NAME).asString());
+        params = extractParamsToMap(resource1);
+        assertFeatureIdParam(SUBSYSTEM, TEST_SUBSYSTEM, params);
+        assertFeatureIdParam(TEST, SPECIAL_NAMES_RESOURCE, params);
+        Assert.assertTrue(params.containsKey("host-feature"));
+        Assert.assertTrue(params.containsKey("test-feature"));
+        Assert.assertTrue(params.containsKey("profile-feature"));
+
+        ModelNode resource2 = feature.require(CHILDREN).require("subsystem.testsubsystem.resource.main-resource");
+        Assert.assertEquals("subsystem.testsubsystem.resource.main-resource", resource2.get(NAME).asString());
+        params = extractParamsToMap(resource2);
+        assertFeatureIdParam(SUBSYSTEM, TEST_SUBSYSTEM, params);
+        assertFeatureIdParam(RESOURCE, MAIN_RESOURCE, params);
+        Assert.assertTrue(params.containsKey("optional-attr"));
+        Assert.assertTrue(params.containsKey("mandatory-attr"));
+
+        // check that nested resource children are also listed
+        Assert.assertTrue(resource2.require(CHILDREN).get("subsystem.testsubsystem.resource.main-resource.list-attr").isDefined());
+        Assert.assertTrue(resource2.require(CHILDREN).get("subsystem.testsubsystem.resource.main-resource.object-attr").isDefined());
+    }
+
+    /**
+     * Reads alias to a storage resource.
+     *
+     * Expectations:
+     *
+     * - the target resource is read.
+     */
+    @Test
+    public void testAliasToResource() throws OperationFailedException {
+        ModelNode result = readFeature(PathAddress.pathAddress("alias", "alias-to-resource"));
+
+        ModelNode feature = result.require(FEATURE);
+        Assert.assertEquals("subsystem.testsubsystem.resource.main-resource", feature.require(NAME).asString());
+        Map<String, ModelNode> params = extractParamsToMap(feature);
+        assertFeatureIdParam(SUBSYSTEM, TEST_SUBSYSTEM, params);
+        assertFeatureIdParam(RESOURCE, MAIN_RESOURCE, params);
+    }
+
+    private ModelNode readFeature(PathAddress address) throws OperationFailedException {
+        return readFeature(address, false);
+    }
+
+    private ModelNode readFeature(PathAddress address, boolean recursive) throws OperationFailedException {
+        ModelNode operation = createOperation(READ_FEATURE_DESCRIPTION_OPERATION, address);
+        operation.get(RECURSIVE).set(recursive);
+        ModelNode result = executeForResult(operation);
+        System.out.println(result);
```

Comment:
Please remove or comment this out. We want to limit disk IO as it adds a lot of burden to our CI servers where in the end disk is a shared resource amongst many VMs.